apps/src/lib/node/ledger/protocol/transactions/ethereum_events/mod.rs

```diff
@@ -18,13 +18,21 @@ use namada::types::vote_extensions::ethereum_events::MultiSignedEthEvent;
 use namada::types::voting_power::FractionalVotingPower;
 
 use super::ChangedKeys;
-use crate::node::ledger::protocol::transactions::utils::{
-    self, get_active_validators,
-};
+use crate::node::ledger::protocol::transactions::utils;
 use crate::node::ledger::protocol::transactions::votes::{
     calculate_new, calculate_updated, write, Votes,
 };
 
+impl utils::GetVoters for [MultiSignedEthEvent] {
```

Comment:
Do we prefer `HashSet` over `BTreeSet`?

Reply:
I generally prefer hashsets, unless there's an obvious reason to pick btreesets. here iteration order doesn't matter, so I went with what james was already using (a hashset)

Reply:
Agree `HashSet` is preferable where ordering isn't important as it communicates that